CMakeLists.txt

```diff
@@ -0,0 +1,106 @@
+#                                                                             #
+#  RTI grants Licensee a license to use, modify, compile, and create          #
+#  derivative works of the software solely for use with RTI Connext DDS.      #
+#  Licensee may redistribute copies of the software provided that all such    #
+#  copies are subject to this license.                                        #
+#  The software is provided "as is", with no warranty of any type, including  #
+#  any warranty for fitness for any purpose. RTI is under no obligation to    #
+#  maintain or support the software.  RTI shall not be liable for any         #
+#  incidental or consequential damages arising out of the use or inability to #
+#  use the software.                                                          #
+#                                                                             #
+###############################################################################
+
+# Load the dependencies of dds-datamodel-utils and rticonnextdds-cmake-utils.
+# The dds-datamodel-utils repo may be provided by setting the variable
+# DDS_DATAMODELS_UTILS_DIR to point to a local copy of this repository.
+# If this is not provided, this macro downloads it as a dependency.
+# This macro alsso loads all necessary files from the dependencies.
+# This sets two variables that may be used in the parent scope:
+#   * datamodels_utils_dir: the path to the datamodels-utils repo
+#   * rti_cmake_utils_dir: the path to the rticonnextdds-cmake-utils repo
+macro (configure_datamodels_dependencies)
+
+    # Get the absolute path for the dds-datamodels-utils repo if it is defined
+    set(datamodels_utils_dir)
+    if (DEFINED DDS_DATAMODELS_UTILS_DIR)
+        get_filename_component(
+            datamodels_utils_dir
+            "${CMAKE_CURRENT_BINARY_DIR}/${DDS_DATAMODELS_UTILS_DIR}"
+            ABSOLUTE)
+    endif()
+
+    if(NOT IS_DIRECTORY ${datamodels_utils_dir})
+        set(datamodels_utils_name_ dds-datamodels-utils)
+
+        include(FetchContent)
+
+        # Set this policy to avoid updating any submodule repositories
+        cmake_policy(SET CMP0097 NEW)
+
+        set(datamodels_utils_dir ${CMAKE_CURRENT_BINARY_DIR}/${datamodels_utils_name_})
+
+        # Fetch the content of the rticonnextdds-cmake-utils repo
+        FetchContent_Declare(${datamodels_utils_name_}
+            GIT_REPOSITORY https://github.com/rticommunity/dds-datamodels-utils.git
+            GIT_TAG main
+            SOURCE_DIR "${datamodels_utils_dir}"
+        )
+        FetchContent_MakeAvailable(${datamodels_utils_name_})
+
+        unset(datamodels_utils_name_)
+    endif()
+    get_filename_component(rti_cmake_utils_dir "${datamodels_utils_dir}/rticonnextdds-cmake-utils" ABSOLUTE)
+
+    list(APPEND CMAKE_MODULE_PATH "${datamodels_utils_dir}/cmake")
+    list(APPEND CMAKE_MODULE_PATH "${rti_cmake_utils_dir}/cmake/Modules")
```

Comment:
The changes assume that `DDS_DATAMODELS_UTILS_DIR`, if set, is passed in as an already absolute path that doesn't require further transformation. Otherwise, I'm not sure that the current version actually "achieves injection" of the dependency, since `CMAKE_CURRENT_BINARY_DIR` [always points to a project's subdirectory](https://cmake.org/cmake/help/latest/variable/CMAKE_CURRENT_BINARY_DIR.html). The current version seems to assume that `DDS_DATAMODELS_UTILS_DIR` is passed in as relative path, which is kind of confusing, since the caller would have to potentially change the value depending on the relative path between this project and the dependency. Much easier to inject an already absolute path.

Plus it allows to simplify the variables and the code a bit.